tests/test_compiler.py

```diff
@@ -206,3 +208,74 @@ def circuit(x: float):
         mlir_str = str(circuit.mlir)
         result_header = "func.func private @circuit(%arg0: tensor<f64>) -> tensor<f64>"
         assert result_header in mlir_str
+
+    def test_jvp(self):
+        """Test that the correct JVP is returned with QJIT."""
+
+        @qml.jit
+        def jvp(params, tangent):
+            def f(x):
+                y = [jnp.sin(x[0]), x[1] ** 2, x[0] * x[1]]
+                return jnp.stack(y)
+
+            return qml.jvp(f, [params], [tangent])
+
+        x = jnp.array([0.1, 0.2])
+        tangent = jnp.array([0.3, 0.6])
+        res = jvp(x, tangent)
+        assert len(res) == 2
+        assert np.allclose(res[0], jnp.array([0.09983342, 0.04, 0.02]))
+        assert np.allclose(res[1], jnp.array([0.29850125, 0.24000006, 0.12]))
+
+    def test_jvp_without_qjit(self):
+        """Test that an error is raised when using JVP without QJIT."""
+
+        def jvp(params, tangent):
+            def f(x):
+                y = [jnp.sin(x[0]), x[1] ** 2, x[0] * x[1]]
+                return jnp.stack(y)
+
+            return qml.jvp(f, [params], [tangent])
+
+        x = jnp.array([0.1, 0.2])
+        tangent = jnp.array([0.3, 0.6])
+
+        with pytest.raises(
+            RuntimeError, match="Pennylane does not support the JVP function without QJIT."
+        ):
+            jvp(x, tangent)
+
+    def test_vjp(self):
+        """Test that the correct VJP is returned with QJIT."""
+
+        @qml.qjit
+        def vjp(params, cotangent):
+            def f(x):
+                y = [jnp.sin(x[0]), x[1] ** 2, x[0] * x[1]]
+                return jnp.stack(y)
+
+            return qml.vjp(f, [params], [cotangent])
+
+        x = jnp.array([0.1, 0.2])
+        dy = jnp.array([-0.5, 0.1, 0.3])
+
+        res = vjp(x, dy)
+        assert len(res) == 2
+        assert np.allclose(res[0], jnp.array([0.09983342, 0.04, 0.02]))
+        assert np.allclose(res[0], jnp.array([-0.43750208, 0.07000001]))
+
+    def test_vjp_without_qjit(self):
+        """Test that an error is raised when using VJP without QJIT."""
+
+        def vjp(params, cotangent):
+            def f(x):
+                y = [jnp.sin(x[0]), x[1] ** 2, x[0] * x[1]]
+                return jnp.stack(y)
+
+            return qml.vjp(f, [params], [cotangent])
+
+        x = jnp.array([0.1, 0.2])
+        dy = jnp.array([-0.5, 0.1, 0.3])
+
+        with pytest.raises(RuntimeError, match="Pennylane does not support the VJP function without QJIT."):
```

Comment:
To fix pylint failure, 
```suggestion
        with pytest.raises(
            RuntimeError, match="Pennylane does not support the VJP function without QJIT."
        ):
```